app/app.go

```diff
@@ -976,6 +980,33 @@ func NewApp(
 	app.SetEndBlocker(app.EndBlocker)
 	app.SetAnteHandler(anteHandler)
 
+	// Create vote extension for signing batches.
+	pvKeyFile := filepath.Join(homePath, cast.ToString(appOpts.Get("priv_validator_key_file")))
+	loadPath := filepath.Join(filepath.Dir(pvKeyFile), utils.SEDAKeyFileName)
+	signer, err := utils.LoadSEDASigner(loadPath)
+	if err != nil {
+		app.Logger().Error("error loading SEDA signer - ExtendVoteHandler will not run", "path", loadPath)
```

Comment:
That sounds good to me, but restarting the process to activate the signer seems inevitable, right?

Reply:
It's probably the easiest, but we could (not saying we should :P) do an approach where we always register the handler and have that be responsible for loading the keyfile. When extending a vote it just checks if it has the keys in memory, and if not it attempts to read the file.

Reply:
We could just have dynamic watching of the file so they don't have to restart? That would mean if that file is edited, changed and saved, it would be reloaded.

Reply:
interesting ideas.. added an issue here https://github.com/sedaprotocol/seda-chain/issues/375